spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -16,26 +16,128 @@
 package org.springframework.kafka.core;
 
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.kafka.clients.producer.Callback;
 import org.apache.kafka.clients.producer.KafkaProducer;
 import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.PartitionInfo;
+
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.context.Lifecycle;
 
 /**
  * @author Gary Russell
  *
  */
-public class DefaultKafkaProducerFactory<K, V> implements ProducerFactory<K, V> {
+public class DefaultKafkaProducerFactory<K, V> implements ProducerFactory<K, V>, Lifecycle, DisposableBean {
+
+	private static final Log logger = LogFactory.getLog(DefaultKafkaProducerFactory.class);
 
 	private final Map<String, Object> configs;
 
+	private volatile Producer<K, V> producer;
+
+	private volatile boolean running;
+
 	public DefaultKafkaProducerFactory(Map<String, Object> configs) {
 		this.configs = new HashMap<>(configs);
 	}
 
 	@Override
+	public void destroy() throws Exception {
+		Producer<K, V> producer = this.producer;
+		this.producer = null;
+		if (producer != null) {
+			producer.close();
```

Comment:
Is it expected?Looking into the `this.producer = new CloseSafeProducer<K, V>()`, and its `no-op` `close()` impl, I'd say that this is a dead code and we don't need this `start()`, `stop()` and `destroy()` "hell".

Or have you meant something like `producer.delegate.close();` ?